src/openfl/text/_internal/ShapeCache.hx

```diff
@@ -0,0 +1,109 @@
+package openfl.text._internal;
+import haxe.ds.IntMap;
+import haxe.ds.StringMap;
+
+#if !openfl_debug
+@:fileXml('tags="haxe,release"')
+@:noDebug
+#end
+
+@:access(openfl.text.TextFormat)
+@SuppressWarnings("checkstyle:FieldDocComment")
+class ShapeCache
```

Comment:
Probably appropriate to give some credit to chromium or blink or whoever

Reply:
If I gave credit to every code author I ever learned from after studying their open source code, it would be a pretty long list. 😝 There's nothing in here that warrants accreditation.